climada/entity/exposures/base.py

```diff
@@ -445,6 +420,8 @@ def set_from_raster(self, file_name, band=1, src_crs=None, window=False,
             resampling (rasterio.warp,.Resampling optional): resampling
                 function used for reprojection to dst_crs
         """
+        if 'geometry' in self.gdf:
+            raise ValueError("there is already a geometry column defined in the GeoDataFrame")
```

Comment:
Fail early, don't read the raster data first.

Reply:
👍